### ISSUE TITLE
All-Time Deltaboard

### ISSUE THREAD

**Issue by Snorrrlax** [MystK/delta-bot-three#34]
Can the all-time deltaboard be filled out by scanning through the full user flair list here?: https://www.reddit.com/r/changemyview/about/flair/

**Comment by MystK** [comment]
Yes that is the trick to support all time deltaboards. It just needs to be implemented.

**Comment by Snorrrlax** [comment]
Ah okay! I guess it depends on #33 being solved first though for it to be accurate.

**Comment by MystK** [comment]
Yes you're right, work can be done in parallel though.

**Comment by Snorrrlax** [comment]
Perhaps this will be achieved in conjunction with the yearly deltaboard scan, @RaoulMeyer? i.e. rather than two separate scans for these two boards, scan the entire subreddit each time and use the whole thing for all-time, but just use "current year + last 6 months of previous year" for yearly?

Is it possible that this could be a work-overload?

**Comment by RaoulMeyer** [comment]
That's definitely a possible solution for this issue. The two boards have a lot in common, so a solution that combines both boards is probably ideal. I don't think the load would be a problem, as this basically would do the same as the year board, but longer.

**Comment by Snorrrlax** [comment]
Thinking about it some more, it seems inefficient to continuously rescan archived posts (those older than 6 months) since new deltas can never be added to those. Why not scan the entire thing once (which would also find all missed deltas), store it in the wiki somehow and build on that for both Deltaboards?

**Comment by wyantb** [comment]
Rather than storing that in the wiki, you could consider just committing a JSON blob of that to the repo, along with the comment id of the last thing scanned.  Assuming we have the disk space available, this would be a lot faster, and you could just update the blob once a month or so to fill in the backlog since the last archived post scan.
